nightfall-client/src/routes/commitment.mjs

```diff
@@ -30,10 +30,12 @@ router.get('/salt', async (req, res, next) => {
   }
 });
 
-router.get('/balance', async (req, res, next) => {
-  logger.debug('commitment/balance endpoint received GET');
+router.post('/balance', async (req, res, next) => {
```

Comment:
Please revert all these `POST` changes. `GET` is correct. I'm failing to see how this is a POST when we are requesting from server on the API level and on database level querying the database too